src/networkextension.cs

```diff
@@ -1109,8 +1114,8 @@ interface NETunnelProvider
 	{
 		[Export ("handleAppMessage:completionHandler:")]
 		[Async]
-		void HandleAppMessage (NSData messageData, [NullAllowed] Action<NSData> completionHandler);
-	
+		void HandleAppMessage (NSData messageData, [NullAllowed] NETunnelAppMessageHandler completionHandler);
```

Comment:
This is a breaking change, so we can't do it.

The correct way to do this would be to add a new method overload with the correct signature (the `NETunnelAppMessageHandler`), and obsolete the old one, but that seems somewhat unnecessary since I believe it's possible to fix the old signature in a backwards compatible way once the generator is fixed.

In other words it might be best to wait until the generator is fixed.